qiskit/ignis/measurement/discriminator/discriminators.py

```diff
@@ -0,0 +1,426 @@
+                expected_states should be a string or a schedule.
+            discriminant: a discriminant from e.g. scikit learn.
+            schedules: The schedules in cal_results or their names. If None
+            all schedules will be used.
+        """
+        if schedules is None:
+            schedules = [_.header.name for _ in cal_results.results]
+
+        # Sanity checks
+        if isinstance(cal_results, list) and isinstance(expected_states, list):
+            if len(cal_results) != len(expected_states):
+                raise QiskitError('Number of input results and assigned '
+                                  'states must be equal.')
+
+        self._expected_state = {}
+        for idx, schedule in enumerate(schedules):
+            if isinstance(schedule, Schedule):
+                name = schedule.name
+            else:
+                name = schedule
+            expected_state = expected_states[idx]
+            self._expected_state[name] = expected_state
+
+        # Used to rescale the xdata qubit by qubit.
+        self._description = None
+        self._standardize = standardize
+        self._scaler = None
+        self._qubit_mask = qubit_mask
+        self._schedules = schedules
+        self._backend_result_list = []
+        self._fitted = False
+
+        if cal_results is not None:
+            if isinstance(cal_results, list):
+                for result in cal_results:
+                    self._backend_result_list.append(result)
+            else:
+                self._backend_result_list.append(cal_results)
+
+        self._xdata = self.get_xdata(self._backend_result_list, schedules)
+        self._ydata = self.get_ydata(self._backend_result_list, schedules)
+
+    def _add_ydata(self, schedule: Union[Schedule, str]):
+        """
+        Adds the expected state of schedule to self._ydata.
+        Args:
+            schedule: schedule or schedule name.
+            Used to get the expected state.
+        """
+        if isinstance(schedule, Schedule):
+            self._ydata.append(self._expected_state[schedule.name])
+        else:
+            self._ydata.append(self._expected_state[schedule])
+
+    def add_data(self, result, refit=True, expected_state=None):
+        """
+        Args:
+            result: the Result obtained from e.g. backend.run().result()
+            recalc: this parameter is irrelevant and only needed for Liskov
+            principle.
+            refit: refit the discriminator if True.
+            expected_state: the expected state of the discriminator.
+        """
+        pass
+
+    @property
+    def expected_states(self):
+        return self._expected_state
+
+    @property
+    def fitted(self):
+        """True if the discriminator has been fitted to calibration data."""
+        return self._fitted
+
+    @abstractmethod
+    def _scale_data(self, xdata: List[List[float]], refit=False):
```

Comment:
Maybe a standard implementation for this should be provided?

Reply:
The reason I did not implement one is to avoid tying `BaseDiscriminationFitter` to `sklearn`. I'm fine with the idea of implement one using `sklearn`. What do you think?

Reply:
I moved the implementation of `_scale_data` to `BaseDiscriminationFitter`.